packages/core/src/storages/key_value_store.ts

```diff
@@ -694,6 +710,17 @@ export class KeyValueStore {
     }
 }
 
+// Function to check if an object is large (you can tweak the size threshold as needed)
+function isLargeObject(obj: any): boolean {
+    const sizeLimit = 1e6; // Adjust this limit to your needs (1MB here)
+    try {
+        const jsonSize = JSON.stringify(obj).length;
+        return jsonSize > sizeLimit;
+    } catch (e) {
+        return false;
```

Comment:
In case the `JSON.stringify()` throws an `Exception`, it is actually more likely it's a "large object" (`JSON.stringify` throwing on large objects is the issue we're trying to solve with this PR).

However, note that `JSON.stringify` can throw because of other reasons too (e.g. trying to serialize a circular structure)